### PR TITLE
[MERGE] hr_timesheet: Fix global UX

### PR DESCRIPTION
**Pull request by IT-Ideas**
### Goal of the PR

This PR is in the continuity of odoo#62360. 

### Details

- Display remaining hours in task sales_line_id (name_get + task form view)
- Remove internal reference from services demo data
- Display sol in project timesheets list if project is billable
- Remove use of non_allow_billable in timesheets as it has been removed from project and task in previous PR (see above)
- Only recompute planned_hours for service product
- Determine the correct SOL for timesheet
- Set the last SOL of customer on timesheet if none is set on task or project
- Allow edition of so_line in timesheet
- Restrict SOL on project to sale lines with a service product
- Use same widget on partner_id many2One than in sale.order (using ranking)
- Remove timesheets table in SO and invoice portal.They were added in previous PR (see above). This introduces the use of links to /my/timesheets/.
- Review portal timesheets (my/timesheets/ and link from orders and invoices)
- Activate group_uom "Units of Measure" on sale_timesheet install
- Hide partner phone number in task form view
- Only determine SOL of task and timesheet if allow_billable=True
- Filter SOL in task so that it matches the SOL of the project SO
- Display red label if remaining hours is negative
- Change SO compute behavior
- Only open SO for salesman on project overview

task-2409761

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_timesheet/models/sale_order.py [hunk 1]
```diff
@@ -78,6 +79,29 @@ class SaleOrderLine(models.Model):
...
+                        'delivery_info': sol.delivery_info
+                    }
+                res[index] = (id, name)
+        return res
+
+    @api.depends('qty_delivered', 'product_uom_qty')
+    def _compute_delivery_info(self):
+        for sol in self:
+            sol.delivery_info = _('Delivered %g / %g') % (
+                float_round(sol.qty_delivered, precision_digits=2) or 0.0,
```
Comment (fmdl) [inline addons/sale_timesheet/models/sale_order.py:102]: Use rounding of uom
`fields.Float.round(sol.qty_delivered, precision_rounding = sol.product_uom_id.rounding)`
and use format_float, to format float in user lang.
https://github.com/odoo/odoo/pull/54407/files#diff-61162ac65633a1c7b054fc83ce1813f1a7984e3169ff36021713ef441f62a208R1488

File: addons/sale_timesheet/models/sale_order.py [hunk 2]
```diff
@@ -78,6 +79,29 @@ class SaleOrderLine(models.Model):
 
     qty_delivered_method = fields.Selection(selection_add=[('timesheet', 'Timesheets')])
     analytic_line_ids = fields.One2many(domain=[('project_id', '=', False)])  # only analytic lines, not timesheets (since this field determine if SO line came from expense)
+    delivery_info = fields.Char(compute='_compute_delivery_info')
+
+    def name_get(self):
+        res = super(SaleOrderLine, self).name_get()
+        if self.env.context.get('with_remaining_hours'):
+            for index, name_get in enumerate(res):
```
Comment (fmdl) [inline addons/sale_timesheet/models/sale_order.py:87]: for perf (don't need filtered for each line)
```python
if self.env.context.get('with_remaining_hours'):
    names = dict(res)
    result = []
    for line in self:
         name = names.get(line.id)
         if line.product_id.service_policy == 'ordered_timesheet':
               name = name = '{name} - {delivery_info}'.format(name=name, delivery_info':line.delivery_info)
         result.append((line.id, name)
     return result
```

Comment (IT-Ideas) [reply addons/sale_timesheet/models/sale_order.py]: 👍

File: addons/sale_timesheet/models/sale_order.py (hunk 1 above)
Comment (IT-Ideas) [reply addons/sale_timesheet/models/sale_order.py]: 👍  I'll use Float.round. It uses the same behind but this is indeed cleaner!
I will not use the rouding of uom even if it is not ideal as for this provides crazy numbers for hours (which is the main purpose of this task).

File: addons/sale_timesheet/__init__.py [hunk 3]
```diff
@@ -4,3 +4,11 @@
 from . import models
 from . import wizard
 from . import report
+
+from odoo import api, SUPERUSER_ID
+
+
+def _sale_timesheet_post_init(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    settings = env['res.config.settings'].create({'group_uom': True})
```
Comment (tivisse) [inline addons/sale_timesheet/__init__.py:13]: Don't create res.config on init just for that. It doesn't a shitload of other stuff. 

Add the group in the implied_ids of base.group_user instead (as done in set_values in res.config)

Comment (tivisse) [reply addons/sale_timesheet/__init__.py:13]: And it can be done in xml on the group directly. No need of a post_init_hook ;)

**Review by tivisse — COMMENTED** [review]
comment

File: addons/sale_timesheet/__init__.py (hunk 3 above)
Comment (IT-Ideas) [reply addons/sale_timesheet/__init__.py]: 👍

**Comment by tivisse** [comment]
@robodoo r+ merge

**Comment by tivisse** [comment]
@robodoo r+

**Pull request merged**